orm/versioning_bucket.go

```diff
@@ -88,46 +128,93 @@ func (b VersioningBucket) Save(db weave.KVStore, model Object) error {
 }
 
 // Update persists the given data object with a new derived version key in the storage.
-func (b VersioningBucket) Update(db weave.KVStore, currentKey VersionedIDRef, data versionedData) (VersionedIDRef, error) {
+// The VersionedIDRef returned won't be nil on success and contains the new version number.
+// The currentKey must be the latest one in usage or an ErrDuplicate is returned.
+func (b VersioningBucket) Update(db weave.KVStore, currentKey VersionedIDRef, data versionedData) (*VersionedIDRef, error) {
```

Comment:
Logic here seems fine, but I am a bit confused by API.

Version is stored in ID, yet versionedData seems to expect it in the value? Or is versionedData an orm.Object (key and value)?

If the second one, then why pass in the version separately, why not just read it from versionedData? Example usage:

1. Load data from db (including currentKey) 
2. Modify some fields in the oject
3. Call b.Update(db, data) which gets the next key

Anyway, if this is correct (even with not-perfect API), fine to merge. I am sure this will be refined when used in the Electorate, so the above mentioned changes can be another PR.

Reply:
I can see the confusion. Would this be better: `Update(db weave.KVStore, id []byte, data versionedData) (*VersionedIDRef, error)` ? 
`versionedData` is not the orm.Object but the payload you would store in the orm.Object.  `VersionedIDRef` serialized is the key of the orm.Object.

The problem with your step3 is that we usually do not store the object ID within the model. But any operation we do has it available anyway. I will integrate with gov to make it more clear.

Reply:
You can merge this PR and then do gov integration in a separate one.

Let's try to do a series of small, short PRs.. make reviewing and merging easier